core/tls-websocket-server/src/certificate_generation.rs

```diff
@@ -0,0 +1,169 @@
+/*
+	Copyright 2021 Integritee AG and Supercomputing Systems AG
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+
+		http://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+
+*/
+#[cfg(all(not(feature = "std"), feature = "sgx"))]
+use crate::sgx_reexport_prelude::*;
+
+use crate::{error::WebSocketError, WebSocketResult};
+use bit_vec::BitVec;
+use chrono::{prelude::*, TimeZone, Utc as TzUtc};
+use core::convert::TryFrom;
+use rcgen::{date_time_ymd, Certificate, CertificateParams, DistinguishedName, DnType};
+use sp_core::{crypto::Pair, ed25519};
+use std::{
+	string::ToString,
+	time::{SystemTime, UNIX_EPOCH},
+	vec,
+	vec::Vec,
+};
+use yasna::models::ObjectIdentifier;
+
+const ED25519: &[u64] = &[1, 3, 101, 112];
+
+/// Create a sel-signed certificate, signed with the Ed25519 private key
+/// Certificate Params are :
```

Comment:
I'm not sure about the certificate parameters. Please check

Reply:
Thanks for adding the comment with the additional information 👍 In order to verify that these parameters make sense, we should talk to Torben. I myself am not familiar with them and don't know the best practices.